src/Components/Post/Post.jsx

```diff
@@ -1,118 +1,215 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./Post.css";
 import { AiOutlineLike, AiFillLike } from "react-icons/ai";
 import { BsBookmark, BsFillBookmarkFill } from "react-icons/bs";
 import { BiComment } from "react-icons/bi";
-import { Menu, Button, MenuButton, MenuList, MenuItem } from "@chakra-ui/react";
+import {
+  Menu,
+  Button,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Heading,
+} from "@chakra-ui/react";
 import { useDisclosure } from "@chakra-ui/react";
 import { EdiPostModal } from "../modal/editModal";
 import { useDispatch, useSelector } from "react-redux";
-import { deletePosts, getUserId } from "../../features/postSlice";
+import { Box } from "@chakra-ui/react";
+import { Avatar, AvatarBadge, AvatarGroup, WrapItem } from "@chakra-ui/react";
+import {
+  deletePosts,
+  getUserId,
+  getSelectedPost,
+  getCommentId,
+} from "../../features/postSlice";
 import { addToBookmark, removeBookmark } from "../../features/bookmarkSlice";
 import { useNavigate } from "react-router-dom";
 import { addToLike, removeLike } from "../../features/postSlice";
+import { CommentModal } from "../modal/commentModal";
+import { deleteUserComment, getAllComments } from "../../features/commentApi";
+import { EditCommentModal } from "../modal/editCommentModal";
+
 const Post = ({ post }) => {
   const { isOpen, onOpen, onClose } = useDisclosure();
   const dispatch = useDispatch();
   const { bookmark } = useSelector((store) => store.bookmark);
   const { token } = useSelector((store) => store.timeline);
```

Comment:
use some sensible variable names in the redux store so that it would be easy to guess which state is being accessed.